specification/keyvault/data-plane/TypeSpec/models.tsp

```diff
@@ -0,0 +1,389 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+/**
+ * Reflects the deletion recovery level currently in effect for secrets in the
+ * current vault. If it contains 'Purgeable', the secret can be permanently
+ * deleted by a privileged user; otherwise, only the system can purge the secret,
+ * at the end of the retention interval.
+ */
+union DeletionRecoveryLevel {
```

Comment:
This should just be an enum which, by default for Azure Core, are extensible enums. This likely won't generate the code we want it to.

Reply:
...and, technically, we don't want SDKs including this anyway. It can and has changed frequently. It's output-only, so the property we may want to declare as just a `string`. I highly doubt the service team will be generating server-side code from this given they already have an old code base and a process for new versions.

Reply:
This should be a `union` not an `enum` if it is an open set of values of which we may know some.

Reply:
@timotheeguerin replied to my Teams post that this changed a couple sprints ago.